XIVComboExpanded/Combos/VPR.cs

```diff
@@ -466,51 +466,59 @@ protected override uint Invoke(uint actionID, uint lastComboMove, float comboTim
                     return VPR.ThirdGeneration;
                 if (gauge.AnguineTribute == maxtribute - 3)
                     return VPR.FourthGeneration;
+                // Even though this deviates from the way it works in PvP, this is likely the more desired behavior to avoid accidentally using First Generation
+                return VPR.Ouroboros;
             }
 
-            // First step, decide whether or not we need to apply debuff
-            if (OriginalHook(VPR.SteelFangs) == VPR.SteelFangs)
+            // Switch case here for optimization, rather than calling OriginalHook in a lot of places.
+            switch (OriginalHook(VPR.SteelFangs))
             {
-                var noxious = FindTargetEffect(VPR.Debuffs.NoxiousGash);
-                if (level >= VPR.Levels.DreadFangs && (noxious is null || noxious?.RemainingTime < 12)) // 12s hopefully means we won't miss anything on a Reawaken window
-                    return VPR.DreadFangs;
+                // Combo step 1, detect presence of debuff
+                case VPR.SteelFangs:
+                    var noxious = FindTargetEffect(VPR.Debuffs.NoxiousGash);
+                    if (level >= VPR.Levels.DreadFangs && (noxious is null || noxious?.RemainingTime < 13)) // 13s hopefully means we won't miss anything on a Reawaken window
```

Comment:
The full Reawaken combo takes a minimum of 13.2s before skill speed (2.2s for Reawaken, 2.0s for each of the Generations, and 3.0s for Ouroboros).  If this case is being hit, then the next action will be the start of a new combo, which means after this is used, a minimum of an additional 7.5s (before speed) is needed _on top of that 13.2s_ before Dread Fangs can be used again.  Meaning if the buff is at less than 20.7s and we use Steel Fangs here, and then the Reawaken combo, you'll drop the debuff before getting back to Dread Fangs (I actually suspect this is why they are increasing at least the duration cap on Noxious, it's far too easy to drop the debuff during the burst phase).

Despite it potentially being less than optimal, I would actually set this to 20s.  Regardless, though, I'd argue that it needs to be a _minimum_ of 14-15s to ensure the debuff doesn't get dropped before Ouroboros.

Reply:
That's the exact math I was looking for! xD

Reply:
🌔

Reply:
#whelp